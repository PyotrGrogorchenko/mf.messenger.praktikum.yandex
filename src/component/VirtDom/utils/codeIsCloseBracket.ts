export const codeIsCloseBracket = (template: LooseObject): boolean => template.record.content.replace(/ /ig, '') === '{%}%}'
