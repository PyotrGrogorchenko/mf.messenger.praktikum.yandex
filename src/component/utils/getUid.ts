export const getUid = (function (): () => number {
  let uidCount: number = 0
  return (): number => {
    uidCount++
    return uidCount
  }
}())
