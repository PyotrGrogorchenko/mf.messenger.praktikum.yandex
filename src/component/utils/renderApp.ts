import { selectRootComponent, setRootComponent } from '../store'

export const renderApp = (Component: any) => {
  const root: HTMLElement | null = document.querySelector('.app')
  if (!root) throw new Error('Element with class="app" not found')
  const app = new Component()
  const RootComponent = selectRootComponent()
  setRootComponent(app)
  if (RootComponent) {
    RootComponent.componentWillUnmount()
    RootComponent.unmount()
    RootComponent.componentDidUnmount()
    while (root.firstChild) {
      root.removeChild(root.firstChild)
    }
  }
  app.init(root)
}
