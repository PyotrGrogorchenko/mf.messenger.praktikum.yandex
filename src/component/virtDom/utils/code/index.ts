import { calculateValue } from './calculateValue'
import { compare } from './compare'
import { isCloseBracket } from './isCloseBracket'

export { calculateValue, compare, isCloseBracket }
