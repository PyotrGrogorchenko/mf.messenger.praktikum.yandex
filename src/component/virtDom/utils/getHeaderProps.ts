import { get, regexpMatchAll } from '@utils'
import { regExpParam } from '../constsnts'

export const getHeaderProps = (context: LooseObject, template: LooseObject, tagName: string): any => {
  let { content } = template.record
  const nodeProps: any = {}

  const prefixCache = '#textCache'
  const cacheTxt: LooseObject = {}
  let count: number = 1
  regexpMatchAll(content, /['"](.*?)['"]/gi).forEach((txt: RegExpExecArray) => {
    if (txt[0]) {
      cacheTxt[`${prefixCache}${count}`] = txt[0].substr(1, txt[0].length - 2)
      content = content.replace(txt[0], `${prefixCache}${count}`)
      count++
    }
  })

  content.split(' ').forEach((keyValue: string) => {
    if (!keyValue || keyValue === tagName) {
      return
    }

    const arrKeyValue: Array<string> = keyValue.split('=')

    if (cacheTxt[arrKeyValue[1]] && arrKeyValue.length > 1) {
      arrKeyValue[1] = cacheTxt[arrKeyValue[1]]
    }

    const param: RegExpExecArray | null = new RegExp(regExpParam).exec(arrKeyValue[1])

    if (arrKeyValue[0] === 'className') {
      const strClasses: string = (param ? get(context, param[1], '') : arrKeyValue[1].replace(/['"]/g, '')).trim()
      if (strClasses) {
        nodeProps.classes = strClasses.split(' ')
      }
    } else if (arrKeyValue.length === 1) {
      nodeProps[arrKeyValue[0]] = '#noValue'
    } else if (param) {
      if (param[1].startsWith(prefixCache)) {
        nodeProps[arrKeyValue[0]] = cacheTxt[param[1]]
      } else {
        nodeProps[arrKeyValue[0]] = get(context, param[1], '')
      }
    } else {
      nodeProps[arrKeyValue[0]] = arrKeyValue[1].replace(/['"]/g, '')
    }
  })

  return nodeProps
}
