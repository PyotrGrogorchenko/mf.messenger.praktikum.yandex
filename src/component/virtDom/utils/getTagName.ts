export const getTagName = (str: string): string => {
  const begin: number = 0
  const end: number = str.indexOf(' ') === -1 ? str.length : str.indexOf(' ')
  return str.slice(begin, end)
}
