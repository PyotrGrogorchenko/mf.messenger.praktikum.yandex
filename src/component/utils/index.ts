import { getUid } from './getUid'
import { copyObj } from './copyObj'
import { get } from './get'
import { regexpMatchAll } from './regexpMatchAll'
import { startsWithUpper } from './startsWithUpper'
import { isEqual } from './isEqual'

export {
  getUid, copyObj, get, regexpMatchAll, isEqual, startsWithUpper
}
