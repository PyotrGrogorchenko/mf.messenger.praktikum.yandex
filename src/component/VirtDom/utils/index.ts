import { isCloseBracket, compare, calculateValue } from './code'
import { inc } from './inc'
import { parent } from './parent'
import { getTagName } from './getTagName'
import { getHeaderProps } from './getHeaderProps'

export {
  isCloseBracket, compare, calculateValue, getHeaderProps,
  inc, parent, getTagName
}
