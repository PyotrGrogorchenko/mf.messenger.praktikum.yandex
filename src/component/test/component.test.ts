// import 'core-js/stable'
import 'regenerator-runtime/runtime'
// import fs from 'fs'
// import path from 'path'
// import { html_beautify } from 'js-beautify'
import { TestElements } from './example/TestElements.gpp'
import { getRoot, writeResult } from './utils'

// const getRoot = () => document.createElement('div')

test('Component => TestElements', async () => {
  const root = getRoot()
  const app = new TestElements()
  app.init(root)
  writeResult('testElements.html', 'template.html', root.innerHTML)
  expect(root.innerHTML).toMatchSnapshot()
})
