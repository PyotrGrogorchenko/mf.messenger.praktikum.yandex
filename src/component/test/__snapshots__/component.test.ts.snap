// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component => TestElements 1`] = `
"<div class=\\"page-column\\" uid=\\"14\\"><form id=\\"form__main\\" class=\\"auth-bar-form\\" uid=\\"15\\"><h3 class=\\"bar__header\\" uid=\\"16\\">Sign up</h3><div class=\\"bar__content\\" uid=\\"17\\"><div class=\\"input\\" uid=\\"18\\"><label class=\\"input__label\\" for=\\"input_first-name\\" uid=\\"19\\">First name</label><input class=\\"input__input\\" type=\\"name\\" id=\\"input_first-name\\" value=\\"first_name\\" uid=\\"20\\"></div><div class=\\"input\\" uid=\\"21\\"><label class=\\"input__label\\" for=\\"input_second-name\\" uid=\\"22\\">Second name</label><input class=\\"input__input\\" type=\\"name\\" id=\\"input_second-name\\" value=\\"second_name\\" uid=\\"23\\"></div><div class=\\"input\\" uid=\\"24\\"><label class=\\"input__label\\" for=\\"input_login\\" uid=\\"25\\">Login</label><input class=\\"input__input\\" type=\\"login\\" id=\\"input_login\\" value=\\"login\\" uid=\\"26\\"></div><div class=\\"input\\" uid=\\"27\\"><label class=\\"input__label\\" for=\\"input_email\\" uid=\\"28\\">email</label><input class=\\"input__input\\" type=\\"mail\\" id=\\"input_email\\" value=\\"email\\" uid=\\"29\\"></div><div class=\\"input\\" uid=\\"30\\"><label class=\\"input__label\\" for=\\"input_password\\" uid=\\"31\\">Password</label><input class=\\"input__input\\" type=\\"password\\" id=\\"input_password\\" value=\\"\\" uid=\\"32\\"></div><div class=\\"input\\" uid=\\"33\\"><label class=\\"input__label\\" for=\\"input_phone\\" uid=\\"34\\">Phone</label><input class=\\"input__input\\" type=\\"phone\\" id=\\"input_phone\\" value=\\"phone\\" uid=\\"35\\"></div></div><div class=\\"bar__footer\\" uid=\\"36\\"><button type=\\"submit\\" class=\\"button-main\\" id=\\"button-sign-up\\" href=\\"props.href\\" uid=\\"37\\">Sign up
      </button><button type=\\"submit\\" class=\\"button-main\\" id=\\"button-to-log-in\\" href=\\"props.href\\" uid=\\"38\\">Log in
      </button></div></form></div>"
`;

exports[`Component 1`] = `
"<div class=\\"page-column\\" uid=\\"14\\"><form id=\\"form__main\\" class=\\"auth-bar-form\\" uid=\\"15\\"><h3 class=\\"bar__header\\" uid=\\"16\\">Sign up</h3><div class=\\"bar__content\\" uid=\\"17\\"><div class=\\"input\\" uid=\\"18\\"><label class=\\"input__label\\" for=\\"input_first-name\\" uid=\\"19\\">First name</label><input class=\\"input__input\\" type=\\"name\\" id=\\"input_first-name\\" value=\\"first_name\\" uid=\\"20\\"></div><div class=\\"input\\" uid=\\"21\\"><label class=\\"input__label\\" for=\\"input_second-name\\" uid=\\"22\\">Second name</label><input class=\\"input__input\\" type=\\"name\\" id=\\"input_second-name\\" value=\\"second_name\\" uid=\\"23\\"></div><div class=\\"input\\" uid=\\"24\\"><label class=\\"input__label\\" for=\\"input_login\\" uid=\\"25\\">Login</label><input class=\\"input__input\\" type=\\"login\\" id=\\"input_login\\" value=\\"login\\" uid=\\"26\\"></div><div class=\\"input\\" uid=\\"27\\"><label class=\\"input__label\\" for=\\"input_email\\" uid=\\"28\\">email</label><input class=\\"input__input\\" type=\\"mail\\" id=\\"input_email\\" value=\\"email\\" uid=\\"29\\"></div><div class=\\"input\\" uid=\\"30\\"><label class=\\"input__label\\" for=\\"input_password\\" uid=\\"31\\">Password</label><input class=\\"input__input\\" type=\\"password\\" id=\\"input_password\\" value=\\"\\" uid=\\"32\\"></div><div class=\\"input\\" uid=\\"33\\"><label class=\\"input__label\\" for=\\"input_phone\\" uid=\\"34\\">Phone</label><input class=\\"input__input\\" type=\\"phone\\" id=\\"input_phone\\" value=\\"phone\\" uid=\\"35\\"></div></div><div class=\\"bar__footer\\" uid=\\"36\\"><button type=\\"submit\\" class=\\"button-main\\" id=\\"button-sign-up\\" href=\\"props.href\\" uid=\\"37\\">Sign up
      </button><button type=\\"submit\\" class=\\"button-main\\" id=\\"button-to-log-in\\" href=\\"props.href\\" uid=\\"38\\">Log in
      </button></div></form></div>"
`;
